qemu/tests/blockdev_mirror_cancel_running_job.py

```diff
@@ -0,0 +1,42 @@
+from provider.blockdev_mirror_nowait import BlockdevMirrorNowaitTest
+from provider.job_utils import get_event_by_condition
+
+
+class BlockdevMirrorCancelRunningJob(BlockdevMirrorNowaitTest):
+    """
+    Cancel a running mirror job
+    """
+
+    def cancel_job(self):
+        self.main_vm.monitor.cmd("job-cancel", {'id': self._jobs[0]})
+        event = get_event_by_condition(
+            self.main_vm, 'BLOCK_JOB_CANCELLED',
+            self.params.get_numeric('job_cancelled_timeout', 60),
+            device=self._jobs[0]
+        )
+        if event is None:
+            self.test.fail('Job failed to cancel')
+
+    def do_test(self):
+        self.blockdev_mirror()
+        self.cancel_job()
```

Comment:
One concern, do we need to check when is the job running?

Reply:
OK, @zhenchao, have added the code to check if job in running status.